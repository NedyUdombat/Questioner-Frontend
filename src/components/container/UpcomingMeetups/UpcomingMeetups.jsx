import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';

// actions
import { getUpcomingMeetups } from '../../../store/modules/meetup';

// config
import { DEFAULT_IMAGE_URL } from '../../../utils/config';

// components
import Image from '../../presentational/Image/Image';

// styles
import './UpcomingMeetups.scss';

export class UpcomingMeetups extends React.Component {
  componentDidMount() {
    this.props.getUpcomingMeetups();
  }

  render() {
    const { meetups } = this.props;
    return (
      <div>
        <div className="d-flex justify-content-center">
          <div className="title-tag">
            <p>Upcoming Meetups</p>
            <div className="d-flex justify-content-center">
              <div className="dash" />
            </div>
          </div>
        </div>
        <div className="section group">
          {meetups.length === 0 ? (
            <p>No Upcoming meetups</p>
          ) : (
            meetups.map(meetup => (
              <div className="col span_1_of_4" key={meetup.id}>
                <div className="d-flex justify-content-center">
                  <div className="product-card">
                    <a href="#">
                      <div className="card-image-top">
                        <Image
                          src={meetup.image || DEFAULT_IMAGE_URL}
                          alt="fr"
                          className="product-image"
                        />
                      </div>
                      <div className="product-card-details">
                        <p className="f-20 ">
                          <b>{meetup.topic}</b>
                        </p>
                        <div className="d-flex justify-content-between mt-1">
                          <small className="f-12">
                            <i>{meetup.organizer_name}</i>
                          </small>
                          <p className="text-blue">
                            {moment(meetup.happening_on).format('DD MMMM YYYY')}
                          </p>
                        </div>
                      </div>
                    </a>
                  </div>
                </div>
              </div>
            ))
          )}
        </div>
      </div>
    );
  }
}

UpcomingMeetups.propTypes = {
  getUpcomingMeetups: PropTypes.func,
  meetups: PropTypes.array,
};
export const mapStateToProps = state => ({
  meetups: state.meetups.meetups,
});

export default connect(
  mapStateToProps,
  { getUpcomingMeetups },
)(UpcomingMeetups);
