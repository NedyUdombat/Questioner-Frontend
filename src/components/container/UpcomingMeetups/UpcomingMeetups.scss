@import '../../../assets/sass/index';

.product-card {
  background-color: $primary-white;
  border-radius: .1875rem;
  height: 21.875rem;
  max-width: 18.75rem;
  min-height: 100%;
  min-width: 15.625rem;
}

.product-card:hover {
  box-shadow: .0625rem .75rem 1rem 0 rgba(0, 0, 0, .24), .0625rem 1.0625rem 3.125rem 0 rgba(0, 0, 0, .19);
  transition: .7s;
}

.product-card .card-image-top {
  height: 70%;
}

.product-card .product-image {
  height: 100%;
  width: 100%;
}

.product-card .product-card-details {
  padding: .625rem;
}

.product-card .product-card-details .d-flex {
  margin-top: .625rem;
}

.title-tag {
  margin-top: .625rem;
}

.title-tag p {
  font-family: $bitter;
  font-size: 2rem;
  margin-bottom: .625rem;
}

.title-tag .dash {
  background-color: $primary-black;
  height: .1875rem;
  width: 30%;
}

.section {
  clear: both;
  margin: 0;
  padding: 3.125rem;
}

.col {
  display: block;
  float: left;
  margin: 1% 0;
}

.group:before,
.group:after {
  content: '';
  display: table;
}

.group:after {
  clear: both;
}

.group {
  zoom: 1;
}

.span_1_of_4 {
  width: 20%;
}

@media only screen and (max-width: 47.5rem) {
  .col {
    margin: 1% 0;
  }

  .span_1_of_4 {
    width: 100%;
  }
}

@media (min-width: 47.5rem) and (max-width: 61.25rem) {
  .col {
    display: block;
    float: left;
    margin: 3% 0;
  }

  .span_1_of_4 {
    width: 50%;
  }

  .section {
    padding: 0;
  }

  .sm .title-tag p {
    display: none;
  }
}

@media (min-width: 61.25rem) and (max-width: 64rem) {
  .col {
    display: block;
    float: left;
    margin: 3% 0;
  }

  .span_1_of_4 {
    width: 30%;
  }

  .section {
    padding: 0;
  }

  .sm .title-tag p {
    display: none;
  }
}
