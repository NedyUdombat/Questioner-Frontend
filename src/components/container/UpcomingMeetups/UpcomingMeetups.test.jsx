import React from 'react';
import { mount } from 'enzyme';

import { UpcomingMeetups, mapStateToProps } from './UpcomingMeetups';

describe('UpcomingMeetups', () => {
  it('should render correctly', () => {
    const props = {
      getUpcomingMeetups: jest.fn(),
      meetups: [],
    };
    const component = mount(<UpcomingMeetups {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with all props', () => {
    const props = {
      getUpcomingMeetups: jest.fn(),
      meetups: [
        {
          id: 'fhwjicgyvufbewf',
          image: 'http://dummyinge.jpg',
          topic: 'topic',
          organizer_name: 'organizer_name',
          happening_on: '2020-05-18T21:28:41+0000',
        },
      ],
    };
    const component = mount(<UpcomingMeetups {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with all props and without image', () => {
    const props = {
      getUpcomingMeetups: jest.fn(),
      meetups: [
        {
          id: 'fhwjicgyvufbewf',
          topic: 'topic',
          organizer_name: 'organizer_name',
          happening_on: '2020-05-18T21:28:41+0000',
        },
      ],
    };
    const component = mount(<UpcomingMeetups {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should mount with the correct state', () => {
    const DEFAULT_STATE = {
      meetups: {
        meetups: [],
      },
    };
    const action = mapStateToProps(DEFAULT_STATE);
    expect(action).toEqual(DEFAULT_STATE.meetups);
  });
});
