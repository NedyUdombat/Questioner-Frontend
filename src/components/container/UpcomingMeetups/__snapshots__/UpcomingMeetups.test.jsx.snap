// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingMeetups should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UpcomingMeetups
      getUpcomingMeetups={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      meetups={Array []}
    >
      <div>
        <div
          className="d-flex justify-content-center"
        >
          <div
            className="title-tag"
          >
            <p>
              Upcoming Meetups
            </p>
            <div
              className="d-flex justify-content-center"
            >
              <div
                className="dash"
              />
            </div>
          </div>
        </div>
        <div
          className="section group"
        >
          <p>
            No Upcoming meetups
          </p>
        </div>
      </div>
    </UpcomingMeetups>
  </Router>
</BrowserRouter>
`;

exports[`UpcomingMeetups should render correctly with all props 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UpcomingMeetups
      getUpcomingMeetups={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      meetups={
        Array [
          Object {
            "happening_on": "2020-05-18T21:28:41+0000",
            "id": "fhwjicgyvufbewf",
            "image": "http://dummyinge.jpg",
            "organizer_name": "organizer_name",
            "topic": "topic",
          },
        ]
      }
    >
      <div>
        <div
          className="d-flex justify-content-center"
        >
          <div
            className="title-tag"
          >
            <p>
              Upcoming Meetups
            </p>
            <div
              className="d-flex justify-content-center"
            >
              <div
                className="dash"
              />
            </div>
          </div>
        </div>
        <div
          className="section group"
        >
          <div
            className="col span_1_of_4"
            key="fhwjicgyvufbewf"
          >
            <div
              className="d-flex justify-content-center"
            >
              <div
                className="product-card"
              >
                <Link
                  to="#"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <div
                      className="card-image-top"
                    >
                      <Image
                        alt="fr"
                        className="product-image"
                        src="http://dummyinge.jpg"
                      >
                        <img
                          alt="fr"
                          className="product-image"
                          src="http://dummyinge.jpg"
                        />
                      </Image>
                    </div>
                    <div
                      className="product-card-details"
                    >
                      <p
                        className="f-20 "
                      >
                        <b>
                          topic
                        </b>
                      </p>
                      <div
                        className="d-flex justify-content-between mt-1"
                      >
                        <small
                          className="f-12"
                        >
                          <i>
                            organizer_name
                          </i>
                        </small>
                        <p
                          className="text-blue"
                        >
                          18 May 2020
                        </p>
                      </div>
                    </div>
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </UpcomingMeetups>
  </Router>
</BrowserRouter>
`;

exports[`UpcomingMeetups should render correctly with all props and without image 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UpcomingMeetups
      getUpcomingMeetups={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      meetups={
        Array [
          Object {
            "happening_on": "2020-05-18T21:28:41+0000",
            "id": "fhwjicgyvufbewf",
            "organizer_name": "organizer_name",
            "topic": "topic",
          },
        ]
      }
    >
      <div>
        <div
          className="d-flex justify-content-center"
        >
          <div
            className="title-tag"
          >
            <p>
              Upcoming Meetups
            </p>
            <div
              className="d-flex justify-content-center"
            >
              <div
                className="dash"
              />
            </div>
          </div>
        </div>
        <div
          className="section group"
        >
          <div
            className="col span_1_of_4"
            key="fhwjicgyvufbewf"
          >
            <div
              className="d-flex justify-content-center"
            >
              <div
                className="product-card"
              >
                <Link
                  to="#"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <div
                      className="card-image-top"
                    >
                      <Image
                        alt="fr"
                        className="product-image"
                        src="https://react.semantic-ui.com/images/wireframe/image.png"
                      >
                        <img
                          alt="fr"
                          className="product-image"
                          src="https://react.semantic-ui.com/images/wireframe/image.png"
                        />
                      </Image>
                    </div>
                    <div
                      className="product-card-details"
                    >
                      <p
                        className="f-20 "
                      >
                        <b>
                          topic
                        </b>
                      </p>
                      <div
                        className="d-flex justify-content-between mt-1"
                      >
                        <small
                          className="f-12"
                        >
                          <i>
                            organizer_name
                          </i>
                        </small>
                        <p
                          className="text-blue"
                        >
                          18 May 2020
                        </p>
                      </div>
                    </div>
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </UpcomingMeetups>
  </Router>
</BrowserRouter>
`;
